src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java

```diff
@@ -0,0 +1,120 @@
+package org.jenkinsci.plugins.plaincredentials;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.fileupload.disk.DiskFileItem;
+import org.apache.commons.fileupload.disk.DiskFileItemFactory;
+import org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl;
+import org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import com.trilead.ssh2.crypto.Base64;
+
+import hudson.security.ACL;
+import hudson.util.Secret;
+
+import static org.junit.Assert.*;
+import static org.hamcrest.CoreMatchers.*;
+
+public class BaseTest {
+
+    private static final String UPDATED_CRED_ID = "Custom-ID-Updated";
+    private static final String CRED_ID = "Custom-ID";
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+    
+    private CredentialsStore store;
+    
+    @Before
+    public void setup(){
+        store = CredentialsProvider.lookupStores(r.jenkins).iterator().next();;
+    }
+    
+    @Test
+    public void secretTextBaseTest() throws IOException {
+        StringCredentialsImpl credential = new StringCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret Text", Secret.fromString("password"));
+        StringCredentialsImpl updatedCredential = new StringCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), credential.getSecret()); 
+        testCreateUpdateDetele(credential, updatedCredential);
+    }
+    
+    @Test
+    public void secretFileBaseTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+        
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret file", fileItem, "keys.txt", Base64.encode(fileItem.get()).toString());
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getData());
+        testCreateUpdateDetele(credential, updatedCredential);
+    }
+    
+    /**
+     * Creates, updates and deletes credentials and perform different assertions
```

Comment:
Not really sure what the point of these assertions is. Are you actually testing any code from this plugin? Or could this all be moved into `credentials-plugin`, for example using `UsernamePasswordCredentialsImpl`?